lib/app.js

```diff
@@ -24,21 +24,89 @@ app.use('/api', ensureAuth);
 // and now every request that has a token in the Authorization header will have a `req.userId` property for us to see who's talking
 app.get('/api/test', (req, res) => {
   res.json({
-    message: `in this proctected route, we get the user's id like so: ${req.userId}`
+    message: `in this protected route, we get the user's id like so: ${req.dessertsId}`
   });
 });
 
-app.get('/animals', async(req, res) => {
+app.post('/desserts', async (req, res) => {
   try {
-    const data = await client.query('SELECT * from animals');
-    
-    res.json(data.rows);
-  } catch(e) {
-    
+    const data = await client.query(`
+    INSERT INTO desserts 
+    (name, icing, type) 
+    VALUES ($1, $2, $3)
+    RETURNING * `, [
+      req.body.name,
+      req.body.icing,
+      req.body.type
+    ]);
+    res.json(data.rows[0]);
+  } catch (e) {
+
     res.status(500).json({ error: e.message });
   }
 });
 
-app.use(require('./middleware/error'));
+app.get('/desserts/:id', async (req, res) => {
+  const id = req.params.id;
+  try {
+    const data = await client.query('SELECT * from desserts/:id', [id]);
+
+    res.json(data.rows[0]);
+  } catch (e) {
+
+    res.status(500).json({ error: e.message });
+  }
+});
+
+app.put('/desserts', async (req, res) => {
+  try {
+    const data = await client.query(`
+    INSERT INTO desserts 
+    (name, icing, type) 
+    VALUES ($1, $2, $3)
+    RETURNING * `, [
+      req.body.name,
+      req.body.icing,
+      req.body.type
+    ]);
+    res.json(data.rows[0]);
+  } catch (e) {
+
+    res.status(500).json({ error: e.message });
+  }
+});
+
+app.post('/desserts/:id', async (req, res) => {
+  try {
+    const data = await client.query(`
+    INSERT INTO desserts 
+    (name, icing, type) 
+    VALUES ($1, $2, $3)
+    RETURNING * `, [
+      req.body.name,
+      req.body.icing,
+      req.body.type
+    ]);
+    res.json(data.rows[0]);
```

Comment:
Nearly there with the POST route! Since you are creating a new entry, you won't know the ID of it yet since SQL adds this automatically for you. This means that you'd call your POST from the /desserts route directly.